### PR TITLE
librbd: ensure compare-and-write doesn't skip compare after copyup

### PR DESCRIPTION
**Pull request by idryomov**

### CODE REVIEW

File: src/include/rados/librados.hpp [hunk 1]
```diff
@@ -303,7 +303,7 @@ namespace librados
     //flag mean ObjectOperationFlags
     void set_op_flags2(int flags);
 
-    void cmpext(uint64_t off, bufferlist& cmp_bl, int *prval);
+    void cmpext(uint64_t off, const bufferlist& cmp_bl, int *prval);
```
Comment (dillaman) [inline src/include/rados/librados.hpp]: Note: the backport PRs should leave the parameter non-const in previous releases.

**Comment by dillaman** [comment]
You should also change [1] so the signatures match.

[1] https://github.com/ceph/ceph/blob/master/src/test/librados_test_stub/LibradosTestStub.cc#L729

**Comment by idryomov** [comment]
@dillaman Yeah, I was unplugged and didn't do a full build.  I think you meant https://github.com/ceph/ceph/blob/master/src/test/librados_test_stub/LibradosTestStub.cc#L821?  The one on line 729 is at a higher level, where they are all non-const (write, write_full, append, etc).

**Comment by dillaman** [comment]
@idryomov Indeed -- thx.

**Comment by dillaman** [comment]
> [ RUN      ] TestLibRBD.CompareAndWriteCopyupMismatch
> using new format!
> /home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/librbd/test_librbd.cc:4583: Failure
>       Expected: -84
> To be equal to: clone_image.compare_and_write(512, write_bl.length(), cmp_bl, write_bl, &mismatch_off, 0)
>       Which is: 512
> /home/jenkins-build/build/workspace/ceph-pull-requests-arm64/src/test/librbd/test_librbd.cc:4600: Failure
> Value of: passed
>   Actual: false
> Expected: true
> [  FAILED  ] TestLibRBD.CompareAndWriteCopyupMismatch (302 ms)
>

**Comment by idryomov** [comment]
It passes on real OSDs.  cmpext() in librados_test_stub is busted, I'll push the fix...

**Comment by dillaman** [comment]
@idryomov thx -- working for me now.

**Pull request merged**